frontend/src/metabase/parameters/components/FilterApplyButton/FilterApplyButton.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useRef } from "react";
+import _ from "underscore";
+import { CSSTransition } from "react-transition-group";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+
+import {
+  getDashboardComplete,
+  getDraftParameterValues,
+  getParameterValues,
+} from "metabase/dashboard/selectors";
+import { setParameterValues } from "metabase/dashboard/actions";
+import {
+  ApplyButton,
+  BUTTON_TRANSITION_DURATION,
+  TRANSITION_CLASSNAMES_PREFIX,
+} from "./FilterApplyButton.styled";
+
+export default function FilterApplyButton() {
+  const isAutoApplyFilters = useSelector(
```

Comment:
Use new `react-redux` hooks

Reply:
We should define selectors in the appropriate file where selectors are located for this feature. For `dashboard` feature folder it's `metabase/dashboard/selectors.js`.

The same with actions - we shouldn't (and we don't) define actions (thunks) inline. Could you move redux action code from `handleApplyFilters` to an action in `metabase/dashboard/actions` folder like in the code snippet I've shared before?

Reply:
That's fair.